base/iobuffer.jl

```diff
@@ -42,7 +42,7 @@ end
 
 # allocate Vector{UInt8}s for IOBuffer storage that can efficiently become Strings
 StringMemory(n::Integer) = unsafe_wrap(Memory{UInt8}, _string_n(n))
-StringVector(n::Integer) = wrap(Array, StringMemory(n))
+StringVector(n::Integer) = view(StringMemory(n), 1:n)::Vector{UInt8}
```

Comment:
I think this change broke JSON3:
```Julia
(base) oscar@Oscars-MBP JSONSchema % julia +nightly
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.12.0-DEV.313 (2024-04-07)
 _/ |\__'_|_|_|\__'_|  |  Commit c5a3b653353 (0 days old master)
|__/                   |

julia> Base.StringVector(UInt64(2))
ERROR: TypeError: in new, expected Tuple{Int64}, got a value of type Tuple{UInt64}
Stacktrace:
 [1] view
   @ ./genericmemory.jl:312 [inlined]
 [2] StringVector(n::UInt64)
   @ Base ./iobuffer.jl:45
 [3] top-level scope
   @ REPL[1]:1

julia> exit()
(base) oscar@Oscars-MBP JSONSchema % julia         
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.10.2 (2024-03-01)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> Base.StringVector(UInt64(2))
2-element Vector{UInt8}:
 0x50
 0x47
```

Example: https://github.com/quinnj/JSON3.jl/blob/6429adafd02a7ad07fa80e9a0ad21b098635244c/src/write.jl#L32

MathOptInterface failing on nightly: https://github.com/jump-dev/MathOptInterface.jl/actions/runs/8591539106/job/23540421403?pr=2464

<img width="1074" alt="image" src="https://github.com/JuliaLang/julia/assets/8177701/67442dae-326f-45ed-89cc-148835faf1ae">

Reply:
Thank you for reporting. I can confirm that this is a bug and this PR introduced it.

Reply:
PR fixing this: #53991